contracts/SinglePlayerCommit.sol

```diff
@@ -51,213 +61,470 @@ contract SinglePlayerCommit is Ownable {
+            console.log("Constructor called for SinglePlayerCommit contract");
+            require(_activityList.length >= 1, "SPC::constructor - activityList empty");
+            token = IERC20(_token);
+
+            _addActivities(_activityList, _oracleAddress);
     }
 
     // view functions
-
-    function getActivityName(bytes32 _activityKey) public view returns (string memory) {
-        return allowedActivities[_activityKey].name;
+    /// @notice Get name string of activity based on key
+    /// @param _activityKey Keccak256 hashed, encoded name of activity
+    /// @dev Lookup in mapping and get name field
+    function getActivityName(bytes32 _activityKey) public view returns (string memory activityName) {
+        return activities[_activityKey].name;
     }
 
     // other public functions
-    function depositAndCommit(
-        bytes32 _activityKey,
-        uint256 _goalValue,
-        uint256 _startTime,
-        uint256 _stake,
-        uint256 _depositAmount
-    ) public returns (bool) {
-        require(deposit(_depositAmount), "SPC::depositAndCommit - deposit failed");
-        require(makeCommitment(_activityKey, _goalValue, _startTime, _stake), "SPC::depositAndCommit - commitment failed");
+    /// @notice Deposit amount of <token> into contract
+    /// @param amount Size of deposit
+    /// @dev Transfer amount to <token> contract, update balance, emit event
+    function deposit(uint256 amount) public returns (bool success) {
+        console.log("Received call for depositing amount %s from sender %s", amount, msg.sender);
+        require(
+            token.transferFrom(msg.sender, address(this), amount), 
+            "SPC::deposit - token transfer failed"
+        );
+
+        _changeCommitterBalance(amount, true);
+
+        emit Deposit(msg.sender, amount);
 
         return true;
     }
 
-    function deposit(uint256 amount) public returns (bool) {
-        console.log("Received call for depositing amount %s from sender %s", amount, msg.sender);
-        // make deposit
-        require(token.transferFrom(msg.sender, address(this), amount), "SPC::deposit - token transfer failed");
+    /// @notice Public function to withdraw unstaked balance of user
+    /// @param amount Amount of <token> to withdraw
+    /// @dev Check balances and active stake, withdraw from balances, emit event
+    function withdraw(uint256 amount) public returns (bool success) {
+        console.log("Received call for withdrawing amount %s from sender %s", amount, msg.sender);
+        uint256 available = committerBalances[msg.sender];
+        Commitment storage commitment = commitments[msg.sender];
 
-        // update committer's balance
-        _changeCommitterBalance(amount, true);
+        if(commitment.exists == true){
+            available = available.sub(commitment.stake);
+        }
 
-        emit Deposit(msg.sender, amount);
+        require(amount >= available, "SPC::withdraw - not enough (unstaked) balance available");
```

Comment:
I think this needs to be `<=` not greater than or equal. The available amount must cover the withdrawal amount.

Reply:
Oopsie. Fixed